lib/mayaHydra/hydraExtensions/adapters/renderItemAdapter.cpp

```diff
@@ -258,42 +285,59 @@ void MayaHydraRenderItemAdapter::UpdateFromDelta(const UpdateFromDeltaData& data
                 }
             }
 
-            // VtArray operator[] is oddly expensive, ~10ms per frame here. Replace with assign().
-            // for (int i = 0; i < indexCount; i++) vertexIndices[i] = indicesData[i];
             vertexIndices.assign(indicesData, indicesData + indexCount);
 
             if (maxIndex < (int64_t)_positions.size() - 1) {
                 _positions.resize(maxIndex + 1);
             }
+            const size_t numNormals = _normals.size();
+            if (numNormals > 0 && (maxIndex < (int64_t)numNormals - 1)) {
+                _normals.resize(maxIndex + 1);
+            }
 
             switch (GetPrimitive()) {
             case MHWRender::MGeometry::Primitive::kTriangles:
                 vertexCounts.resize(indexCount / 3);
                 vertexCounts.assign(indexCount / 3, 3);
 
-                // UVs
+                // Get face varying data from Maya like uvs
                 if (indexCount > 0) {
+                    
                     MVertexBuffer* mvb = nullptr;
-                    for (int vbIdx = 0; vbIdx < geom->vertexBufferCount(); vbIdx++) {
+
+                    for (int vbIdx = 0; vbIdx < vertexBuffercount; vbIdx++) {
                         mvb = geom->vertexBuffer(vbIdx);
                         if (!mvb)
                             continue;
 
                         const MVertexBufferDescriptor& desc = mvb->descriptor();
-
-                        if (desc.semantic() != MGeometry::Semantic::kTexture)
-                            continue;
-
-                        // Hydra expects a uv coordinate for each face-index, not 1 per vertex. 
-                        // e.g. a cube expects 36 uvs not 24.
-                        _uvs.clear();
-                        _uvs.resize(indices->size());
-                        float* uvs = (float*)mvb->map();
-                        for (int i = 0; i < indexCount; ++i) {
-                            _uvs[i].Set(&uvs[indicesData[i] * 2]);
+                        const auto semantic = desc.semantic();
+                        switch(semantic){
+                            case MGeometry::Semantic::kTexture: {
+                                // Hydra supports a uv coordinate for each face-index (face varying), though we could use its own set of indices which should be smaller.
+                                _uvs.clear();
+                                _uvs.resize(indices->size());
+                                float* uvs = (float*)mvb->map();
+                                for (int i = 0; i < indexCount; ++i){
+                                    _uvs[i].Set(&uvs[indicesData[i] * 2]);
+                                }
+                                mvb->unmap();
+                            }
+                            break;
+                            case MHWRender::MGeometry::kTangent:{
```

Comment:
Didn't see notion for tangent type in Hydra, is mikkt tangent not supported in Hydra, right?
Also, it's good to have a test case to cover tangent.

Reply:
Yes only Mikkt is supported for tangents, but it's not explicitly mentioned in Hydra.
I left the code for tangents for a later PR but we need to modify the translation of materials to support tangents. At this time the tangents won't be used so I cannot add a test for them.